STYLEGUIDE.md

```diff
@@ -0,0 +1,260 @@
+
+    Returns:
+        bool
+    """
+```
+
+## <a name="dict">Dictionary Key-Value Retrieval</a>
+When retrieving a value from a dictionary key, do not use square bracket notation. This method will return an error if the 
+indicated key does not exist in the dictionary. Instead, use the `get` method which returns `None` by default if the 
+indicated key does not exist in the dictionary.
+
+**Don't:**
+```python
+bucket_name = desired_state_definition["Bucket"]
+```
+**Do:**
+```python
+# dict.get("Key", default=None)
+bucket_name = desired_state_definition.get("Bucket")
+```
+
+For testing if a dictionary returns empty, use the following notation:
+```python
+bucket_name = desired_state_definition.get("Bucket", {})
+```
+
+## <a name="pcf_util">PCF Utility Fucntions</a>
+There are several functions in <a href="https://github.com/capitalone/Particle-Cloud-Framework/blob/master/pcf/util/pcf_util.py">pcf_util.py</a>
+that perform various tasks such as creating a dictionary passed on a key set and finding nested values in a dictionary.
+Refer to this module when performing such complex operations. If there is not an existing function that meets your needs, simply
+create one. The desired functions are then imported into the particular module you are implementing.
+
+Example:
+```python
+def param_filter(curr_dict, key_set, remove=False):
+    """
+    Filters param dictionary to only have keys in the key set
+    Args:
+        curr_dict (dict): param dictionary
+        key_set (set): set of keys you want
+        remove (bool): filters by what to remove instead of what to keep
+    Returns:
+        filtered param dictionary
+    """
+    if remove:
+        return {key: curr_dict[key] for key in curr_dict.keys() if key not in key_set}
+    else:
+        return {key: curr_dict[key] for key in key_set if key in curr_dict.keys()}
+```
+
+## <a name="pcf_excep">PCF Exceptions</a>
+There are several exceptions in <a href="https://github.com/capitalone/Particle-Cloud-Framework/blob/master/pcf/core/pcf_exceptions.py">pcf_exceptions.py</a>
+that define various exceptions. Refer to this module when considering exception handling. If there is not an existing exception that meets your needs, simply
+create one. The desired exceptions are then imported into the particular module you are implementing.
+
+Example:
+```python
+class NoCodeException(Exception):
+    def __init__(self):
+        Exception.__init__(self, "Did not provide local zipfile or zipfile location in S3")
+```
+
+## <a name="logging">Logging Standards</a>
+Logging is a means of tracking events that happen when some software runs. More information on logging in Python can be found <a href="https://docs.python.org/3/howto/logging.html#logging-basic-tutorial">here</a>.
+
+To enable logging in PCF, add the following code to the top of your pcf python file.
+
+```python
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+for handler in logging.root.handlers:
```

Comment:
Why are we using filters? Not sure we need this. Might be nice to instantiate a logger with the class name though.